test-interface/src/main/scala/scala/scalanative/testinterface/TestMainBase.scala

```diff
@@ -148,3 +194,29 @@ abstract class TestMainBase {
     override def trace(t: Throwable): Unit = log(Log.Level.Trace, "", Some(t))
   }
 }
+
+object TestMainBase {
+  private val faultHandler = CFunctionPtr.fromFunction1(handleSegFault _)
+  private def handleSegFault(id: Int): Unit = {
+    //making ids stable i.e. vals instead fo defs
+    //for use in the match
+    val SIGSEGV = signal.SIGSEGV
+    val SIGFPE  = signal.SIGFPE
+
+    val retCode = id match {
+      case `SIGSEGV` =>
+        Console.err.println("Segmentation fault")
+        139
+      case `SIGFPE` =>
+        Console.err.println("Erroneous Arithmetic Operation")
+        136
```

Comment:
I don't think these values are enforced: http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/signal.h.html

My understanding is that the exit code should be `1 << 7 + id`. The first bit is set to indicate an abnormal exit, the following 7 bits are represent the exit code.

Reply:
I did not know that. Will change it to the formula.